### PR TITLE
runtime: print error when panicking with error interface type

### PR DESCRIPTION
**Pull request by aykevl**

### CODE REVIEW

File: src/runtime/print.go [hunk 1]
```diff
@@ -201,6 +201,8 @@ func printitf(msg interface{}) {
 	switch msg := msg.(type) {
 	case string:
 		print(msg)
+	case error:
+		print(msg.Error())
 	default:
```
Comment (johanbrandhorst) [inline src/runtime/print.go]: Thought, do we want to add a `case fmt.Stringer`?

Comment (aykevl) [reply src/runtime/print.go]: Good idea, `panic()` in the Go runtime also seems to accept that: https://play.golang.org/p/UOeakJdGX6_3

**Comment by aykevl** [comment]
I have added some more cases, so that booleans, integers, and Stringer gets printed as well. This increases code size in some cases but it's not by a large margin and it is very useful for debugging. Also, if code size is a concern, it can be disabled using https://github.com/tinygo-org/tinygo/pull/299.

**Comment by deadprogram** [comment]
Excellent work here. Merging.

**Pull request merged**